tools/autograd/derivatives.yaml

```diff
@@ -302,6 +302,10 @@
 - name: atan2(Tensor self, Tensor other) -> Tensor
   self, other: atan2_backward(grad, self, other, grad_input_mask)
 
+- name: attn(Tensor q, Tensor k, Tensor v) -> (Tensor, Tensor)
+  q, k: attn_backward(grads[0], grads[1], q, k, v, result1)
+  v: 'grads[0].defined() ? result1.t().matmul(grads[0]) : zeros_like(v)'
```

Comment:
For gradients, an undefined Tensor is the same as a Tensor full of zeros. So the second case here can be `Tensor()` and that will be much more efficient.

Reply:
easy enough.